File: frontend/lib/models/family_member_model.dart
class FamilyMember {
  final String id;
  final String userId;
  final String familyMemberId;
  final String name;
  final String email;
  final String relation;
  final String status; // Pending, Accepted, Rejected

  FamilyMember({
    required this.id,
    required this.userId,
    required this.familyMemberId,
    required this.name,
    required this.email,
    required this.relation,
    required this.status,
  });

  factory FamilyMember.fromJson(Map<String, dynamic> json) {
    final member = json['familyMember'] ?? {};
    return FamilyMember(
      id: json['id'].toString(),
      userId: json['userId'].toString(),
      familyMemberId: json['familyMemberId'].toString(),
      name: member['fullName'] ?? '',
      email: member['email'] ?? '',
      relation: json['relation'] ?? '',
      status: json['status'] ?? 'Pending',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'familyMemberId': familyMemberId,
      'relation': relation,
      'status': status,
    };
  }
}

File: backend/src/models/family_member.model.js
const { DataTypes } = require("sequelize");
const { sequelize } = require("../config/db");

const FamilyMember = sequelize.define(
  "FamilyMember",
  {
    id: {
      type: DataTypes.BIGINT.UNSIGNED,
      autoIncrement: true,
      primaryKey: true,
    },
    userId: {
      type: DataTypes.BIGINT.UNSIGNED,
      allowNull: false,
      references: {
        model: "users",
        key: "id",
      },
    },
    familyMemberId: {
      type: DataTypes.BIGINT.UNSIGNED,
      allowNull: false,
      references: {
        model: "users",
        key: "id",
      },
    },
    relation: {
      type: DataTypes.STRING(50),
      allowNull: false,
      validate: {
        notEmpty: { msg: "Relation is required." },
      },
    },
    status: {
      type: DataTypes.ENUM("Pending", "Accepted", "Rejected"),
      allowNull: false,
      defaultValue: "Pending",
    },
  },
  {
    tableName: "family_members",
    timestamps: true,
    createdAt: "createdAt",
    updatedAt: "updatedAt",
  }
);

module.exports = FamilyMember;